editor/project_manager.cpp

```diff
@@ -1165,10 +1165,13 @@ void ProjectManager::_load_recent_projects() {
 
 	bool set_ordered_latest_modification;
 	ProjectListFilter::FilterOption filter_order_option = project_order_filter->get_filter_option();
-	if (filter_order_option == ProjectListFilter::FILTER_NAME)
+	if (filter_order_option == ProjectListFilter::FILTER_NAME) {
 		set_ordered_latest_modification = false;
-	else
+		EditorSettings::get_singleton()->set("default_projects_order", "Name");
```

Comment:
To do this cleanly (as it will be exposed in the Editor Settings), I would put it in a `project_manager` category, like `project_manager/sorting_order` (no need to mention `default`, it's not just a default as you can't override it - if you change it, the default is changed too :)).

And I would save the enum value instead of a string. You can do it with:
```
EditorSettings::get_singleton()->add_property_hint(PropertyInfo(Variant::INT, "project_manager/sorting_order", PROPERTY_HINT_ENUM, "Name,Last Modified"));
```